### PR TITLE
Update web socket to handle binary data.

### PR DESCRIPTION
**Pull request by kendalharland**
- Add improved tests for WebSocket
- Make the public interface more consistent with other WebSocket implementations:
  - Change 'onData' to the more commonly understood 'onMessage'
  - Open a websocket on construction instead of using a method call
  - Expose one-shot streams to detect when the socket opens/closes
  - Wrap payloads in SeltzerMessages to give clients more info on received mssages.

### CODE REVIEW

File: lib/platform/browser.dart
Comment (matanlurey) [inline lib/platform/browser.dart]: nit: Type as Complete<Null> (here and below)

File: lib/platform/browser.dart
Comment (matanlurey) [inline lib/platform/browser.dart]: Stream<Null> here and below.

File: lib/platform/browser.dart
Comment (matanlurey) [inline lib/platform/browser.dart]: Future<Null> (here and below)

File: lib/platform/browser.dart
Comment (matanlurey) [inline lib/platform/browser.dart]: Use the types to make it shorter:

``` dart
static Future<_BrowserSeltzerMessage> create(MessageEvent message) ...
```

File: lib/platform/server.dart
Comment (matanlurey) [inline lib/platform/server.dart:52]: Same as the comments in the other class.

File: lib/src/interface.dart
Comment (matanlurey) [inline lib/src/interface.dart:257]: Hmm, maybe we should just have a:

``` dart
ByteBuffer readAsBytes();
String readAsString();
```

And just snip `payload`.

**Review by matanlurey — APPROVED** [review]
Looks really great, just a few comments.

File: lib/platform/server.dart
Comment (kendalharland) [reply lib/platform/server.dart:52]: done

File: lib/src/interface.dart
Comment (kendalharland) [reply lib/src/interface.dart:257]: Done.  Added asynchronous callbacks because I can't find a way to parse a dart:html Blob synchronously.  Subclasses can always provide their own reasAsXSync methods if desired.

**Comment by kendalharland** [comment]
Let's wait until tomorrow to merge this. I want to work through [this guide](https://docs.travis-ci.com/user/gui-and-headless-browsers/) and fix content-shell for this package.  It's annoying to not have the ability to run browsers tests while I'm working on cloud9 and also not be able to trust Travis results

**Comment by kendalharland** [comment]
I updated the browser implementation and tested locally. Everything looks good.

**Pull request merged**
